src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,158 @@
+                                        address(this),
+                                        address(this),
+                                        eigenLayerShareBIPs,
+                                        rootPostDelay
+                                    )
+                                )));
+        
+        dummyToken = new ERC20Mock();
+    }
+
+    function testInitialize() public {
+        require(paymentCoordinator.rootPublisher() == address(this), "rootPublisher should be set");
+        require(paymentCoordinator.merkleRootActivationDelay() == 7200, "merkleRootActivationDelay should be set");
+        require(paymentCoordinator.eigenLayerShareBIPs() == 1000, "eigenLayerShareBIPs should be set");
+        require(paymentCoordinator.owner() == address(this), "owner should be set");
+    }
+
+    function testMakePayment(uint256[] memory amounts) public {
+        cheats.assume(amounts.length > 0);
+        uint256 sum = _sum(amounts);
+        PaymentCoordinator.Payment memory payment;
+        payment.token = dummyToken;
+        payment.amounts = amounts;
+
+        uint256 balanceBefore = dummyToken.balanceOf(address(paymentCoordinator));
+        uint256 cumumlativeEarningsBefore = paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken);
+        cheats.expectEmit(address(paymentCoordinator));
+        emit PaymentReceived(address(this), payment);
+        paymentCoordinator.makePayment(payment);
+        require(paymentCoordinator.cumulativeEigenLayerTokeEarnings(dummyToken) - cumumlativeEarningsBefore == sum * eigenLayerShareBIPs / MAX_BIPS, "cumulativeEigenLayerTokeEarnings should be set");
+        require(dummyToken.balanceOf(address(paymentCoordinator)) - balanceBefore == sum, "incorrect token balance");
+    }
+
+    function testPostMerkleRootFromUnauthorizedAddress(address unauthorizedAddress) external {
+        cheats.assume(unauthorizedAddress != paymentCoordinator.rootPublisher());
+        cheats.startPrank(unauthorizedAddress);
+        cheats.expectRevert(bytes("PaymentCoordinator: Only rootPublisher"));
+        paymentCoordinator.postMerkleRoot(bytes32(0), 0, 0);
+        cheats.stopPrank();
+    }
+
+    function testPostMerkleRoot(bytes32 root, uint256 height, uint256 calculatedUpToBlockNumber) public {
+        PaymentCoordinator.MerkleRootPost memory merkleRootPost;
+        merkleRootPost.root = root;
+        merkleRootPost.height = height;
+        merkleRootPost.calculatedUpToBlockNumber = calculatedUpToBlockNumber;
+
+        uint256 numMerkleRootsBefore = paymentCoordinator.merkleRootsLength();
+
+        cheats.startPrank(paymentCoordinator.rootPublisher());
+        paymentCoordinator.postMerkleRoot(root, height, calculatedUpToBlockNumber);
+        cheats.stopPrank();
+
+        (bytes32 rootPost, uint256 heightPost, uint256 confirmedAtBlockNumber, uint256 calculatedUpToBlockNumberPost) = paymentCoordinator.merkleRoots(numMerkleRootsBefore);
```

Comment:
this line seems excessively long, recommend breaking it up

Reply:
fixed by returning a struct